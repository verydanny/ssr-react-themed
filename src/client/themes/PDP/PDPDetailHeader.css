@value ( RentBlue ) from 'ui/shared/themes/colors/colors.css';

.PDPDetailHeader {
  height: 66px;
  background-color: RentBlue;
  display: flex;
}

.PDPHeader_Logo, .PDPDetailHeader_Back {
  margin-bottom: 15px;
}

.PDPDetailHeader_Back {
  margin-left: 19px;
  padding: 0;
  border: 0;
  background-color: RentBlue;
}

.PDPDetailHeader_Title {
  color: white;
  margin: 12px 0 0 25px;
}

.PDPDetailHeader_Title > h3 { 
  margin: 0;
  margin-bottom: 3px;
  color: white;
}

.PDPDetailHeader_Title > p { 
  margin: 0;
  font-weight: 300;
  font-size: 14px;
}
