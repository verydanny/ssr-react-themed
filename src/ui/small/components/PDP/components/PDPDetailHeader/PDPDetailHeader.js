import React from 'react'
import PropTypes from 'prop-types'
import { themed } from 'react-themed-too'
import { Icon } from 'ui/shared/components/Icon'
import truncateString from 'lib/utils/truncateString'

const PDPDetailHeader = ({ title, subtitle, onClick, theme, titleMax, subtitleMax }) => {
  const renderText = (text, limit) => {
    if (!limit || text.length <= limit) return text

    return `${truncateString(text, limit)} ...`
  }

  return (
    <div
      className={theme.PDPDetailHeader}
      data-tid="pdpHeader"
    >
      <button
        className={theme.PDPDetailHeader_Back}
        onClick={onClick}
        data-tid="pdpDetailHeader_Back"
      >
        <Icon
          name="arrowLeft"
        />
      </button>

      <div className={theme.PDPDetailHeader_Title}>
        <h3>{renderText(title, titleMax)}</h3>
        {
          subtitle &&
          <p>{renderText(subtitle, subtitleMax)}</p>
        }
      </div>
    </div>
  )
}

PDPDetailHeader.propTypes = {
  theme: PropTypes.object,
  onClick: PropTypes.func,
  title: PropTypes.string.isRequired,
  subtitle: PropTypes.string,
  titleMax: PropTypes.number,
  subtitleMax: PropTypes.number,
}

PDPDetailHeader.defaultProps = {
  theme: {},
}

export default themed(/^PDPDetailHeader/, { pure: true })(PDPDetailHeader)
