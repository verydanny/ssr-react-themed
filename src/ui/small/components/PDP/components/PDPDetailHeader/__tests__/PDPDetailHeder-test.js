import React from 'react'
import { mount } from 'enzyme'
import PDPDetailHeader from '../PDPDetailHeader'

describe('<PDPDetailHeader/>', () => {
  const props = {
    theme: {},
    title: '$1,375+ Malino',
    subtitle: 'Floorplan A4',
    onClick: jest.fn(),
  }

  it('renders a title, subtitle, and back button', () => {
    const wrapper = mount(
      <PDPDetailHeader {...props} />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('renders a title without a subtitle', () => {
    const wrapper = mount(
      <PDPDetailHeader theme={{}} title={'A Detail Header Title'} />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('trims title and subtitle with max characters', () => {
    const wrapper = mount(
      <PDPDetailHeader
        {...props}
        title={'This is a super long name of an address on the detail header'}
        subtitle={'841 Southeast Memorial Day Drive SE, Atlanta, GA'}
        titleMax={30}
        subtitleMax={35}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
  it('calls onClick() when clicked', () => {
    const wrapper = mount(
      <PDPDetailHeader {...props} />
    )
    const button = wrapper.find('button')
    button.simulate('click')
    expect(props.onClick).toHaveBeenCalled()
  })
})
