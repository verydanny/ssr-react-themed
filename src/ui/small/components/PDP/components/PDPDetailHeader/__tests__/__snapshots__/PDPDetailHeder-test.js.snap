// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PDPDetailHeader/> renders a title without a subtitle 1`] = `
<Themed(PDPDetailHeader)
  theme={Object {}}
  title="A Detail Header Title"
>
  <PDPDetailHeader
    theme={Object {}}
    title="A Detail Header Title"
  >
    <div
      data-tid="pdpHeader"
    >
      <button
        data-tid="pdpDetailHeader_Back"
      >
        <Themed(Icon)
          name="arrowLeft"
        >
          <Icon
            name="arrowLeft"
            svgs={Object {}}
            theme={Object {}}
          />
        </Themed(Icon)>
      </button>
      <div>
        <h3>
          A Detail Header Title
        </h3>
      </div>
    </div>
  </PDPDetailHeader>
</Themed(PDPDetailHeader)>
`;

exports[`<PDPDetailHeader/> renders a title, subtitle, and back button 1`] = `
<Themed(PDPDetailHeader)
  onClick={[MockFunction]}
  subtitle="Floorplan A4"
  theme={Object {}}
  title="$1,375+ Malino"
>
  <PDPDetailHeader
    onClick={[MockFunction]}
    subtitle="Floorplan A4"
    theme={Object {}}
    title="$1,375+ Malino"
  >
    <div
      data-tid="pdpHeader"
    >
      <button
        data-tid="pdpDetailHeader_Back"
        onClick={[MockFunction]}
      >
        <Themed(Icon)
          name="arrowLeft"
        >
          <Icon
            name="arrowLeft"
            svgs={Object {}}
            theme={Object {}}
          />
        </Themed(Icon)>
      </button>
      <div>
        <h3>
          $1,375+ Malino
        </h3>
        <p>
          Floorplan A4
        </p>
      </div>
    </div>
  </PDPDetailHeader>
</Themed(PDPDetailHeader)>
`;

exports[`<PDPDetailHeader/> trims title and subtitle with max characters 1`] = `
<Themed(PDPDetailHeader)
  onClick={[MockFunction]}
  subtitle="841 Southeast Memorial Day Drive SE, Atlanta, GA"
  subtitleMax={35}
  theme={Object {}}
  title="This is a super long name of an address on the detail header"
  titleMax={30}
>
  <PDPDetailHeader
    onClick={[MockFunction]}
    subtitle="841 Southeast Memorial Day Drive SE, Atlanta, GA"
    subtitleMax={35}
    theme={Object {}}
    title="This is a super long name of an address on the detail header"
    titleMax={30}
  >
    <div
      data-tid="pdpHeader"
    >
      <button
        data-tid="pdpDetailHeader_Back"
        onClick={[MockFunction]}
      >
        <Themed(Icon)
          name="arrowLeft"
        >
          <Icon
            name="arrowLeft"
            svgs={Object {}}
            theme={Object {}}
          />
        </Themed(Icon)>
      </button>
      <div>
        <h3>
          This is a super long name of ...
        </h3>
        <p>
          841 Southeast Memorial Day Drive SE ...
        </p>
      </div>
    </div>
  </PDPDetailHeader>
</Themed(PDPDetailHeader)>
`;
